ChangeLog.d/driver-only-ecdh.txt

```diff
@@ -0,0 +1,4 @@
+Features
+   * When a PSA driver for ECDH is present, it is now possible to disable
+     MBEDTLS_ECDH_C in the build in order to save code size. For PK, X.509
+     and TLS to fully work, this requires MBEDTLS_USE_PSA_CRYPTO to be enabled.
```

Comment:
PK and X.509 don't use ECDH, and TLS 1.3 does not observe `USE_PSA` when it comes to ECDH, so I think the last sentence should be "For TLS 1.2 key exchanges based on ECDH(E) to work, this requires MBEDTLS_USE_PSA_CRYPTO". I think we also want to add "Restartable/interruptible ECDHE operations in TLS 1.2 (ECDHE-ECDSA key exchange) are not supported in those builds yet, as PSA does not have an API for restartable ECDH yet."